pkg/operation/botanist/component/istio/networkpolicies.go

```diff
@@ -0,0 +1,511 @@
+				return func() error {
+					obj.Annotations = map[string]string{
+						v1beta1constants.GardenerDescription: fmt.Sprintf("Allows Egress from pods labeled with "+
+							"'%s=%s' to DNS running in '%s'.", v1beta1constants.LabelApp, v1beta1constants.DefaultIngressGatewayAppLabelValue,
+							metav1.NamespaceSystem),
+					}
+					obj.Spec = networkingv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								v1beta1constants.LabelApp: istiodAppLabelValue,
+							},
+						},
+
+						Egress: []networkingv1.NetworkPolicyEgressRule{{
+							To: []networkingv1.NetworkPolicyPeer{
+								{
+									NamespaceSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											v1beta1constants.LabelRole: metav1.NamespaceSystem,
+										},
+									},
+									PodSelector: &metav1.LabelSelector{
+										MatchExpressions: []metav1.LabelSelectorRequirement{{
+											Key:      coredns.LabelKey,
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{coredns.LabelValue},
+										}},
+									},
+								},
+								{
+									NamespaceSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											v1beta1constants.LabelRole: metav1.NamespaceSystem,
+										},
+									},
+									PodSelector: &metav1.LabelSelector{
+										MatchExpressions: []metav1.LabelSelectorRequirement{{
+											Key:      coredns.LabelKey,
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{nodelocaldns.LabelValue},
+										}},
+									},
+								},
+							},
+							Ports: []networkingv1.NetworkPolicyPort{
+								{Protocol: protocolPtr(corev1.ProtocolUDP), Port: intStrPtr(coredns.PortServiceServer)},
+								{Protocol: protocolPtr(corev1.ProtocolTCP), Port: intStrPtr(coredns.PortServiceServer)},
+								{Protocol: protocolPtr(corev1.ProtocolUDP), Port: intStrPtr(coredns.PortServer)},
+								{Protocol: protocolPtr(corev1.ProtocolTCP), Port: intStrPtr(coredns.PortServer)},
+							},
+						}},
+						PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+					}
+
+					if values.DNSServerAddress != nil {
+						obj.Spec.Egress[0].To = append(obj.Spec.Egress[0].To, networkingv1.NetworkPolicyPeer{
+							IPBlock: &networkingv1.IPBlock{
+								// required for node local dns feature, allows egress traffic to CoreDNS
+								CIDR: fmt.Sprintf("%s/32", *values.DNSServerAddress),
+							},
+						})
+					}
+
+					if values.NodeLocalIPVSAddress != nil {
+						obj.Spec.Egress[0].To = append(obj.Spec.Egress[0].To, networkingv1.NetworkPolicyPeer{
+							IPBlock: &networkingv1.IPBlock{
+								// required for node local dns feature, allows egress traffic to node local dns cache
+								CIDR: fmt.Sprintf("%s/32", *values.NodeLocalIPVSAddress),
+							},
+						})
+					}
+
+					return nil
+				}
+			},
+		},
+		{
+			name: "allow-from-shoot-vpn",
```

Comment:
I've seen in a tcpdump that istiod gets connections from shoot-vpn pod. If I remove the network policy, the seed doesn't get ready:
```
Message:               error during apply of object "networking.istio.io/v1alpha3/EnvoyFilter/istio-ingress/proxy-protocol": Put "https://api.d046309.core.internal.dev.k8s.ondemand.com:443/apis/networking.istio.io/v1alpha3/namespaces/istio-ingress/envoyfilters/proxy-protocol": context deadline exceeded
```

Reply:
Yes, there is a validating webhook.

Reply:
Ok, that explains it. This should probably be added as a comment or in the description. Otherwise, we will wonder about this next time we look at it...